From 1b20f713f1428363a518500598e5548d210ea7ac Mon Sep 17 00:00:00 2001
From: lanshh <lsh@rock-chips.com>
Date: Tue, 10 Apr 2018 11:35:50 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3326-evb-ai-va-v10

1. add lsensor and psensor
2. remove unused sensors
3. add unprepare-delay-ms up to 40 for panel device
4. add bt sco audio support

Change-Id: Iedde0567f0cf4eaa099f7323e812a58f88c58d1e
Signed-off-by: lanshh <lsh@rock-chips.com>
---
 .../dts/rockchip/rk3326-evb-ai-va-v10.dts     | 80 +++++++++++++------
 1 file changed, 56 insertions(+), 24 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts
index 34217813ad16..3344bd61d71e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-ai-va-v10.dts
@@ -169,6 +169,26 @@
 		};
 	};
 
+	bt-sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "dsp_b";
+		simple-audio-card,bitclock-inversion = <1>;
+		simple-audio-card,mclk-fs = <256>;
+		simple-audio-card,name = "rockchip,bt";
+		simple-audio-card,cpu {
+			sound-dai = <&i2s2_2ch>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&bt_sco>;
+		};
+	};
+
+	bt_sco: bt-sco {
+		compatible = "delta,dfbmcs320";
+		#sound-dai-cells = <0>;
+		status = "okay";
+	};
+
 	rk_headset {
 		compatible = "rockchip_headset";
 		headset_gpio = <&gpio3 RK_PA4 GPIO_ACTIVE_HIGH>;
@@ -245,7 +265,7 @@
 		init-delay-ms = <20>;
 		enable-delay-ms = <120>;
 		disable-delay-ms = <50>;
-		unprepare-delay-ms = <20>;
+		unprepare-delay-ms = <40>;
 
 		width-mm = <68>;
 		height-mm = <121>;
@@ -592,17 +612,6 @@
 	i2c-scl-rising-time-ns = <275>;
 	i2c-scl-falling-time-ns = <16>;
 
-	sensor@0f {
-		status = "okay";
-		compatible = "ak8963";
-		reg = <0x0f>;
-		type = <SENSOR_TYPE_COMPASS>;
-		irq_enable = <0>;
-		poll_delay_ms = <30>;
-		layout = <1>;
-		reprobe_en = <1>;
-	};
-
 	gt1x: gt1x@14 {
 	      compatible = "goodix,gt1x";
 	      reg = <0x14>;
@@ -610,18 +619,6 @@
 	      goodix,irq-gpio = <&gpio0 RK_PA5 IRQ_TYPE_LEVEL_LOW>;
 	};
 
-	sensor@4c {
-		status = "okay";
-		compatible = "gs_mma7660";
-		reg = <0x4c>;
-		type = <SENSOR_TYPE_ACCEL>;
-		irq-gpio = <&gpio0 RK_PB7 IRQ_TYPE_LEVEL_LOW>;
-		irq_enable = <0>;
-		poll_delay_ms = <30>;
-		layout = <1>;
-		reprobe_en = <1>;
-	};
-
 	is31fl3236: led-controller@3c {
 		compatible = "issi,is31fl3236";
 		reg = <0x3c>;
@@ -954,6 +951,35 @@
 			linux,default-trigger = "default-on";
 		};
 	};
+
+	ls_stk3410: light@48 {
+		compatible = "ls_stk3410";
+		status = "okay";
+		reg = <0x48>;
+		type = <SENSOR_TYPE_LIGHT>;
+		irq_enable = <0>;
+		als_threshold_high = <100>;
+		als_threshold_low = <10>;
+		als_ctrl_gain = <2>; /* 0:x1 1:x4 2:x16 3:x64 */
+		poll_delay_ms = <100>;
+	};
+
+	ps_stk3410: proximity@48 {
+		compatible = "ps_stk3410";
+		status = "okay";
+		reg = <0x48>;
+		type = <SENSOR_TYPE_PROXIMITY>;
+		//pinctrl-names = "default";
+		//pinctrl-0 = <&gpio2_c3>;
+		//irq-gpio = <&gpio0 RK_PB7 IRQ_TYPE_LEVEL_LOW>;
+		//irq_enable = <1>;
+		ps_threshold_high = <0x200>;
+		ps_threshold_low = <0x100>;
+		ps_ctrl_gain = <3>; /* 0:x1 1:x4 2:x16 3:x64 */
+		ps_led_current = <3>; /* 0:12.5mA 1:25mA 2:50mA 3:100mA */
+		poll_delay_ms = <100>;
+	};
+
 };
 
 &i2c2 {
@@ -968,6 +994,12 @@
 		     &i2s1_2ch_sdo>;
 };
 
+&i2s2_2ch {
+	status = "okay";
+	rockchip,bclk-fs = <64>;
+	#sound-dai-cells = <0>;
+};
+
 &io_domains {
 	status = "okay";
 
-- 
2.35.3

